examples/vl3_basic/vl3-nse/cmd/service_registry.go

```diff
@@ -0,0 +1,152 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"github.com/sirupsen/logrus"
+	"github.com/tiswanso/examples/api/serviceregistry"
+	"google.golang.org/grpc"
+	"strconv"
+	"strings"
+)
+
+const (
+	POD_NAME = "podName"
+	SERVICE_NAME = "service"
+	PORT = "port"
+	CLUSTER_NAME = "clusterName"
+)
+
+type validationErrors []error
+
+type ServiceRegistry interface {
+	RegisterWorkload(clusterName, podName, name, seviceName, connDom string, ipAddr []string, ports []int32) error
+	RemoveWorkload(clusterName, podName, name, seviceName, connDom string, ipAddr []string, ports []int32) error
+}
+
+type ServiceRegistryClient interface {
+	Stop()
+}
+
+type ServiceRegistryImpl struct {
+	registryClient serviceregistry.RegistryClient
+	connection *grpc.ClientConn
+}
+
+func (s *ServiceRegistryImpl) RegisterWorkload(clusterName, podName, name, seviceName, connDom string, ipAddr []string, ports []int32) error {
+	workloadIdentifier := &serviceregistry.WorkloadIdentifier{
+		Cluster:             clusterName,
+		PodName:             podName,
+		Name:                name,
+	}
+
+	workload := &serviceregistry.Workload{
+		Identifier:          workloadIdentifier,
+		IPAddress:           ipAddr,
+	}
+
+	workloads := []*serviceregistry.Workload{workload}
+	serviceWorkload := &serviceregistry.ServiceWorkload{
+		ServiceName:         seviceName,
+		ConnectivityDomain:  connDom,
+		Workloads:           workloads,
+		Ports:               ports,
+	}
+
+	logrus.Infof("Sending workload register request: %v", serviceWorkload)
+	_, err := s.registryClient.RegisterWorkload(context.Background(), serviceWorkload)
+	if err != nil {
+		logrus.Errorf("service registration not successful: %v", err)
+	}
+
+	return err
+}
+
+func (s *ServiceRegistryImpl) RemoveWorkload(clusterName, podName, name, seviceName, connDom string, ipAddr []string, ports []int32) error {
+	workloadIdentifier := &serviceregistry.WorkloadIdentifier{
+		Cluster:             clusterName,
+		PodName:             podName,
+		Name:                name,
+	}
+
+	workload := &serviceregistry.Workload{
+		Identifier:          workloadIdentifier,
+		IPAddress:           ipAddr,
+	}
+
+	workloads := []*serviceregistry.Workload{workload}
+	serviceWorkload := &serviceregistry.ServiceWorkload{
+		ServiceName:         seviceName,
+		ConnectivityDomain:  connDom,
+		Workloads:           workloads,
+		Ports:               ports,
+	}
+
+	logrus.Infof("Sending workload remove request: %v", serviceWorkload)
+	_, err := s.registryClient.RemoveWorkload(context.Background(), serviceWorkload)
+	if err != nil {
+		logrus.Errorf("service removal not successful: %v", err)
+	}
+
+	return err
+}
+
+func (s *ServiceRegistryImpl) Stop() {
+	s.connection.Close()
+}
+
+func NewServiceRegistry(addr string) (ServiceRegistry, ServiceRegistryClient, error) {
```

Comment:
Can be placed right before serviceRegistry stuct to keep things more organized

Reply:
done